diff -ur qtscriptgenerator-git/generator/parser/rpp/pp-internal.h qtscriptgenerator-git-build/generator/parser/rpp/pp-internal.h
--- qtscriptgenerator-git/generator/parser/rpp/pp-internal.h	2009-05-16 14:09:48.000000000 +0100
+++ qtscriptgenerator-git-build/generator/parser/rpp/pp-internal.h	2009-05-16 14:37:01.000000000 +0100
@@ -16,6 +16,7 @@
 #define PP_INTERNAL_H
 
 #include <algorithm>
+#include <stdio.h>
 
 namespace rpp {
 
