diff --git a/src/context/plasma/animator.cpp b/src/context/plasma/animator.cpp
index 4f386b3..240dc0c 100644
--- a/src/context/plasma/animator.cpp
+++ b/src/context/plasma/animator.cpp
@@ -701,7 +701,7 @@ void AnimatorPrivate::init(Animator *q)
 
     if (!pluginName.isEmpty()) {
         QString constraint = QString("[X-KDE-PluginInfo-Name] == '%1'").arg(pluginName);
-        KService::List offers = KServiceTypeTrader::self()->query("Plasma/Animator", constraint);
+        KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Animator", constraint);
 
         if (!offers.isEmpty()) {
             QString error;
diff --git a/src/context/plasma/applet.cpp b/src/context/plasma/applet.cpp
index 6723b2e..8ba97fe 100644
--- a/src/context/plasma/applet.cpp
+++ b/src/context/plasma/applet.cpp
@@ -606,7 +606,7 @@ QString Applet::category(const QString &appletName)
     }
 
     QString constraint = QString("[X-KDE-PluginInfo-Name] == '%1'").arg(appletName);
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Applet", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Applet", constraint);
 
     if (offers.isEmpty()) {
         return QString();
@@ -1371,7 +1371,7 @@ KPluginInfo::List Applet::listAppletInfo(const QString &category,
         }
     }
 
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Applet", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Applet", constraint);
     //kDebug() << "Applet::listAppletInfo constraint was '" << constraint
     //         << "' which got us " << offers.count() << " matches";
     return KPluginInfo::fromServices(offers);
@@ -1381,7 +1381,7 @@ KPluginInfo::List Applet::listAppletInfoForMimetype(const QString &mimetype)
 {
     QString constraint = QString("'%1' in [X-Plasma-DropMimeTypes]").arg(mimetype);
     //kDebug() << "listAppletInfoForMimetype with" << mimetype << constraint;
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Applet", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Applet", constraint);
     return KPluginInfo::fromServices(offers);
 }
 
@@ -1395,7 +1395,7 @@ QStringList Applet::listCategories(const QString &parentApp, bool visibleOnly)
         constraint.append(" and [X-KDE-ParentApp] == '").append(parentApp).append("'");
     }
 
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Applet", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Applet", constraint);
     QStringList categories;
     foreach (const KService::Ptr &applet, offers) {
         QString appletCategory = applet->property("X-KDE-PluginInfo-Category").toString();
@@ -1425,13 +1425,13 @@ Applet *Applet::load(const QString &appletName, uint appletId, const QVariantLis
     }
 
     QString constraint = QString("[X-KDE-PluginInfo-Name] == '%1'").arg(appletName);
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Applet", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Applet", constraint);
 
     bool isContainment = false;
     if (offers.isEmpty()) {
         //TODO: what would be -really- cool is offer to try and download the applet
         //      from the network at this point
-        offers = KServiceTypeTrader::self()->query("Plasma/Containment", constraint);
+        offers = KServiceTypeTrader::self()->query("AmarokContext/Containment", constraint);
         isContainment = true;
         if (offers.isEmpty()) {
             kDebug() << "offers is empty for " << appletName;
diff --git a/src/context/plasma/containment.cpp b/src/context/plasma/containment.cpp
index fea415f..c35ee07 100644
--- a/src/context/plasma/containment.cpp
+++ b/src/context/plasma/containment.cpp
@@ -823,7 +823,7 @@ KPluginInfo::List Containment::listContainments(const QString &category,
         }
     }
 
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Containment", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Containment", constraint);
     //kDebug() << "constraint was" << constraint << "which got us" << offers.count() << "matches";
     return KPluginInfo::fromServices(offers);
 }
@@ -832,7 +832,7 @@ KPluginInfo::List Containment::listContainmentsForMimetype(const QString &mimety
 {
     QString constraint = QString("'%1' in [X-Plasma-DropMimeTypes]").arg(mimetype);
     //kDebug() << mimetype << constraint;
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/Containment", constraint);
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/Containment", constraint);
     return KPluginInfo::fromServices(offers);
 }
 
diff --git a/src/context/plasma/dataenginemanager.cpp b/src/context/plasma/dataenginemanager.cpp
index 8547be9..3c307db 100644
--- a/src/context/plasma/dataenginemanager.cpp
+++ b/src/context/plasma/dataenginemanager.cpp
@@ -118,7 +118,7 @@ Plasma::DataEngine *DataEngineManager::loadEngine(const QString &name)
 
     // load the engine, add it to the engines
     QString constraint = QString("[X-KDE-PluginInfo-Name] == '%1'").arg(name);
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/DataEngine",
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/DataEngine",
                                                               constraint);
     QString error;
 
@@ -168,7 +168,7 @@ void DataEngineManager::unloadEngine(const QString &name)
 QStringList DataEngineManager::listAllEngines()
 {
     QStringList engines;
-    KService::List offers = KServiceTypeTrader::self()->query("Plasma/DataEngine");
+    KService::List offers = KServiceTypeTrader::self()->query("AmarokContext/DataEngine");
     foreach (const KService::Ptr &service, offers) {
         QString name = service->property("X-KDE-PluginInfo-Name").toString();
         if (!name.isEmpty()) {
diff --git a/src/context/plasma/package.cpp b/src/context/plasma/package.cpp
index 2c26170..8a779be 100644
--- a/src/context/plasma/package.cpp
+++ b/src/context/plasma/package.cpp
@@ -395,7 +395,7 @@ bool Package::registerPackage(const PackageMetadata &data, const QString &iconPa
     KConfigGroup cg = config.desktopGroup();
     const QString type = data.type().isEmpty() ? "Service" : data.type();
     cg.writeEntry("Type", type);
-    const QString serviceTypes = data.serviceType().isNull() ? "Plasma/Applet,Plasma/Containment" : data.serviceType();
+    const QString serviceTypes = data.serviceType().isNull() ? "AmarokContext/Applet,AmarokContext/Containment" : data.serviceType();
     cg.writeEntry("X-KDE-ServiceTypes", serviceTypes);
     cg.writeEntry("X-KDE-PluginInfo-EnabledByDefault", true);
 
diff --git a/src/context/plasma/tests/packagemetadatatest.cpp b/src/context/plasma/tests/packagemetadatatest.cpp
index b657e86..c2a4865 100644
--- a/src/context/plasma/tests/packagemetadatatest.cpp
+++ b/src/context/plasma/tests/packagemetadatatest.cpp
@@ -63,7 +63,7 @@ void PackageMetadataTest::read()
 
     QCOMPARE(pm->name(), QString("Package metadata test file"));
     QCOMPARE(pm->description(), QString("A test desktop file to test the PackageMetaData class."));
-    QCOMPARE(pm->serviceType(), QString("Plasma/Applet"));
+    QCOMPARE(pm->serviceType(), QString("AmarokContext/Applet"));
     QCOMPARE(pm->author(), QString("Bertjan Broeksema"));
     QCOMPARE(pm->email(), QString("b.broeksema@kdemail.net"));
     QCOMPARE(pm->version(), QString("pre0.1"));
@@ -78,7 +78,7 @@ void PackageMetadataTest::write()
 {
     pm->setName(QString("Package metadata test file copy"));
     pm->setDescription(QString("Some other fancy test description"));
-    pm->setServiceType(QString("Plasma/Applet"));
+    pm->setServiceType(QString("AmarokContext/Applet"));
     pm->setAuthor(QString("Bertjan Broeksema"));
     pm->setEmail(QString("b.broeksema@kdemail.net"));
     pm->setVersion(QString("pre0.1"));
@@ -94,7 +94,7 @@ void PackageMetadataTest::write()
 
     QCOMPARE(pm->name(), QString("Package metadata test file copy"));
     QCOMPARE(pm->description(), QString("Some other fancy test description"));
-    QCOMPARE(pm->serviceType(), QString("Plasma/Applet"));
+    QCOMPARE(pm->serviceType(), QString("AmarokContext/Applet"));
     QCOMPARE(pm->author(), QString("Bertjan Broeksema"));
     QCOMPARE(pm->email(), QString("b.broeksema@kdemail.net"));
     QCOMPARE(pm->version(), QString("pre0.1"));
diff --git a/src/context/plasma/tests/packagemetadatatest.desktop b/src/context/plasma/tests/packagemetadatatest.desktop
index 755b2e2..03cd94d 100644
--- a/src/context/plasma/tests/packagemetadatatest.desktop
+++ b/src/context/plasma/tests/packagemetadatatest.desktop
@@ -96,7 +96,7 @@ Comment[zh_TW]=測試 PackageMetaData 類別的桌面檔
 
 Icon=test
 Type=Service
-X-KDE-ServiceTypes=Plasma/Applet
+X-KDE-ServiceTypes=AmarokContext/Applet
 
 X-KDE-Screenshot=a_not_default_file.svg
 X-KDE-Library=some_test_library
